graphql/resolvers/_RootMutations/submitTestimonial.js

```diff
@@ -0,0 +1,163 @@
+
+  if (!testimonial && !image) {
+    console.error('a new testimonials requires an image', args)
+    throw new Error(t('api/testimonial/image/required'))
+  }
+
+  // block if testimonial has a video
+  if (testimonial && testimonial.video) {
+    console.error('testimonial has a video, change not allowed', args)
+    throw new Error(t('api/unexpected'))
+  }
+
+  const firstMembership = await pgdb.public.memberships.findFirst({userId: req.user.id}, {orderBy: ['sequenceNumber asc']})
+  let seqNumber
+  if (firstMembership) { seqNumber = firstMembership.sequenceNumber }
+
+  if (!image) {
+    testimonial = await pgdb.public.testimonials.updateAndGetOne({id: testimonial.id}, {
+      role,
+      quote,
+      published: true,
+      sequenceNumber: testimonial.sequenceNumber || seqNumber
+    }, {skipUndefined: true})
+  } else { // new image
+    const inputBuffer = Buffer.from(image, 'base64')
+    const id = testimonial ? testimonial.id : uuid()
+
+    const pathOriginal = `/${FOLDER}/${id}_original.jpeg`
+    const pathSmall = `/${FOLDER}/${id}_${IMAGE_SIZE_SMALL}x${IMAGE_SIZE_SMALL}.jpeg`
+    const pathShare = `/${FOLDER}/${id}_${IMAGE_SIZE_SHARE}x${IMAGE_SIZE_SHARE}.jpeg`
+
+    await Promise.all([
+      convertImage.toJPEG(inputBuffer)
+        .then((data) => {
+          return uploadExoscale({
+            stream: data,
+            path: pathOriginal,
+            mimeType: 'image/jpeg',
+            bucket: S3BUCKET
+          })
+        }),
+      convertImage.toSmallBW(inputBuffer)
+        .then((data) => {
+          return uploadExoscale({
+            stream: data,
+            path: pathSmall,
+            mimeType: 'image/jpeg',
+            bucket: S3BUCKET
+          })
+        }),
+      convertImage.toShare(inputBuffer)
+        .then((data) => {
+          return uploadExoscale({
+            stream: data,
+            path: pathShare,
+            mimeType: 'image/jpeg',
+            bucket: S3BUCKET
+          })
+        })
+    ])
+
+    if (testimonial) {
+      await keyCDN.purgeUrls([pathOriginal, pathSmall, pathShare])
+      testimonial = await pgdb.public.testimonials.updateAndGetOne({id: testimonial.id}, {
+        role,
+        quote,
+        image: ASSETS_BASE_URL + pathSmall,
+        updatedAt: new Date(),
+        published: true,
+        sequenceNumber: testimonial.sequenceNumber || seqNumber
+      }, {skipUndefined: true})
+    } else {
+      testimonial = await pgdb.public.testimonials.insertAndGet({
+        id,
+        userId: req.user.id,
+        role,
+        quote,
+        image: ASSETS_BASE_URL + pathSmall,
+        published: true,
+        sequenceNumber: seqNumber
+      }, {skipUndefined: true})
+    }
+  }
+
+  // generate sm picture (PNG!)
+  try {
+    const smImagePath = `/${FOLDER}/sm/${testimonial.id}_sm.png`
+    await renderUrl(`${FRONTEND_BASE_URL}/community?share=${testimonial.id}`, 1200, 628)
```

Comment:
are we going to continue doing those share images with phantom cloud?

IMO drop this code for now or disable it.